src/dataset-builder/CohortEditor.ts

```diff
@@ -592,48 +600,141 @@ export const CohortEditor: React.FC<CohortEditorProps> = (props) => {
+    return {
+      chart: {
+        spacingLeft: 20,
+        spacingRight: 30,
+        height: cohortDemographics.height,
+        style: { fontFamily: 'inherit' },
+        type: 'bar',
+      },
+      legend: { enabled: cohortDemographics.legendEnabled },
+      plotOptions: { series: { stacking: 'normal' } },
+      series: cohortDemographics.series,
+      title: {
+        align: 'left',
+        style: { fontSize: '16px', fontWeight: 'bold', color: '#333f52' },
+        text: cohortDemographics.title,
+      },
+      tooltip: {
+        followPointer: true,
+        formatter() {
+          // @ts-ignore
+          // eslint-disable-next-line react/no-this-in-sfc
+          const currCategory = _.find((category) => category.short === this.x, cohortDemographics.categories);
+          const categoryDescription = currCategory.long || currCategory.short;
+          if (cohortDemographics.showSeriesName) {
+            // @ts-ignore
+            // eslint-disable-next-line react/no-this-in-sfc
+            return `${categoryDescription} <br/><span style="color:${this.color}">\u25CF</span> ${this.series.name}<br/> ${this.y}`;
+          }
+          // @ts-ignore
+          // eslint-disable-next-line react/no-this-in-sfc
+          return `${categoryDescription} <br/> ${this.y}`;
+        },
+      },
+      xAxis: {
+        categories: _.map('short', cohortDemographics.categories),
+        crosshair: true,
+      },
+      yAxis: {
+        crosshair: true,
+        title: { text: cohortDemographics.yTitle },
+      },
+      accessibility: {
+        point: {
+          descriptionFormatter: (point) => {
+            return `${point.index + 1}. Category ${point.category}, ${point.series.name}: ${point.y}.`;
+          },
+        },
+      },
+      exporting: { buttons: { contextButton: { x: -15 } } },
+    };
+  }
 
   const updateCohort = (updateCohort: (Cohort) => Cohort) => setCohort(updateCohort);
 
-  return h(Fragment, [
-    h(CohortEditorContents, {
-      updateCohort,
-      cohort,
-      snapshotId,
-      snapshotBuilderSettings,
-      onStateChange,
-      getNextCriteriaIndex,
-    }),
-    // add div to cover page to footer
-    div(
-      {
-        style: {
-          display: 'flex',
-          backgroundColor: editorBackgroundColor,
-          alignItems: 'end',
-          flexDirection: 'row-reverse',
-          padding: wideMargin,
+  useEffect(() => {
+    setSnapshotRequestParticipantCount(
+      withErrorReporting(`Error fetching snapshot builder count for snapshot ${snapshotId}`)(async () =>
+        DataRepo()
+          .snapshot(snapshotId)
+          .getSnapshotBuilderCount(createSnapshotBuilderCountRequest([cohort]))
+      )
+    );
+  }, [snapshotId, setSnapshotRequestParticipantCount, cohort]);
+
+  useEffect(() => {
+    setTimeout(() => {
+      cohortAgeData.series = generateAgeSeries(3, 90);
+      setCohortAges(cohortAgeData);
+    }, 2000);
```

Comment:
I'm not sure this timeout is doing everything I want it to do. It does stop a double re-render (like without it, the graphs re-render when the async count method is called and again when it is returned), but I wanted to add a timeout so that it would return at a similar time as the async count call and it isn't doing that even if I increase the time. What am I missing here?

Reply:
So, I wouldn't expect you to need both the timeout and the useEffect subscribed to snapshotRequestParticipantCount. This effect should be called whenever `snapshotRequestParticipantCount` is updated. The trick might be that `snapshotRequestParticipantCount` is a `useLoadedData` hook, so you probably need to check the status of `useLoadedData` and only display/update when it is `'Ready'`